addons/website/models/website.py

```diff
@@ -479,6 +479,14 @@ def set_images(images):
         theme_name = kwargs['theme_name']
         theme = request.env['ir.module.module'].search([('name', '=', theme_name)])
         url = theme.button_choose_theme()
+        # button_choose_theme redirects the user with
+        # with_loading and enable_editor but if the
+        # user is on a translated page (e.g /en)
+        # the editor never starts (as there's no edit button)
+        # thus never removes the loading animation
+        # so we redirect the user to the default 
+        # language so the editor can initialize properly
+        url['url'] = '/website/lang/default?%s' % urls.url_encode({'r': url['url']})
```

Comment:
should be fixed at the root cause ;) and not only where it causes problem for your use case...
-> the url generation...
```diff
 addons/website/models/website.py | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git addons/website/models/website.py addons/website/models/website.py
index 63bf05fea348..ac921af4c41e 100644
--- addons/website/models/website.py
+++ addons/website/models/website.py
@@ -479,14 +479,6 @@ class Website(models.Model):
         theme_name = kwargs['theme_name']
         theme = request.env['ir.module.module'].search([('name', '=', theme_name)])
         url = theme.button_choose_theme()
-        # button_choose_theme redirects the user with
-        # with_loading and enable_editor but if the
-        # user is on a translated page (e.g /en)
-        # the editor never starts (as there's no edit button)
-        # thus never removes the loading animation
-        # so we redirect the user to the default
-        # language so the editor can initialize properly
-        url['url'] = '/website/lang/default?%s' % urls.url_encode({'r': url['url']})

         # Force to refresh env after install of module
         assert request.env.registry is registry()
@@ -1287,6 +1279,8 @@ class Website(models.Model):
     def button_go_website(self, path='/', mode_edit=False):
         self._force()
         if mode_edit:
+            # force default lang
+            path = url_for(path, self.default_lang_id.code)
             path += '?enable_editor=1'
         return {
             'type': 'ir.actions.act_url',

```